WIP/calculator.rb

```diff
@@ -0,0 +1,95 @@
+class Calculate
+
+  def initialize(**args)
+    @accumulate = args[:accumulate] || args[:accumulate] == nil
+    @discounts = {}
+  end
+
+  def add_discount(**args)
+
+    name = args[:name]
+    amount = args[:amount]
```

Comment:
need to validate that the amount is greater than 0

Reply:
It's validated just below that.If amount is smaller than zero a standard error will be raised.